src/CMakeLists.txt

```diff
@@ -33,12 +33,15 @@ add_library(nats SHARED ${SOURCES} ${PS_SOURCES} ${S_SOURCES})
 target_link_libraries(nats ${NATS_OPENSSL_LIBS} ${NATS_EXTRA_LIB} ${NATS_PROTOBUF_LIB})
 add_library(nats_static STATIC ${SOURCES} ${PS_SOURCES} ${S_SOURCES})
 target_link_libraries(nats_static ${NATS_OPENSSL_LIBS} ${NATS_PROTOBUF_LIB})
+set_target_properties(nats nats_static PROPERTIES
+  VERSION ${NATS_VERSION_MAJOR}.${NATS_VERSION_MINOR}.${NATS_VERSION_PATCH}
+  SOVERSION ${NATS_VERSION_MAJOR}.${NATS_VERSION_MINOR})
 
 # --------------------------------------
 # Install the libraries and header files
 # --------------------------------------
-install(TARGETS nats DESTINATION lib)
-install(TARGETS nats_static DESTINATION lib)
-install(FILES deprnats.h DESTINATION include RENAME nats.h)
-install(FILES nats.h status.h version.h DESTINATION include/nats)
-install(FILES adapters/libevent.h adapters/libuv.h DESTINATION include/nats/adapters)
+install(TARGETS nats LIBRARY DESTINATION ${NATS_LIBDIR})
```

Comment:
This does not work on Windows. See (https://cmake.org/cmake/help/v3.0/command/install.html):
```
For DLL platforms the DLL part of a shared library is treated as a RUNTIME target
and the corresponding import library is treated as an ARCHIVE target.
All Windows-based systems including Cygwin are DLL platforms. 
```
So either you replace this line with:
```
if(WIN32)
  install(TARGETS nats RUNTIME DESTINATION ${NATS_LIBDIR})
else(WIN32)
  install(TARGETS nats LIBRARY DESTINATION ${NATS_LIBDIR})
endif(WIN32)
```
or simply remove `LIBRARY` and simply have:
```
install(TARGETS nats DESTINATION ${NATS_LIBDIR})
```
I tested and it still installs the shared library on macOS, Linux and Windows. If it does not work on your system and you need to specifically have `LIBRARY`, then you need the `ifdef` as shown above.